src/reason_toolchain.ml

```diff
@@ -300,14 +305,29 @@ module Create_parse_entrypoint (Toolchain_impl: Toolchain_spec) :Toolchain = str
 end
 
 module OCaml_syntax = struct
+  open Migrate_parsetree
   module Lexer_impl = Lexer
   module Parser_impl = Parser
 
-  let implementation = Parser.implementation Lexer.token
-  let core_type = Parser.parse_core_type Lexer.token
-  let interface = Parser.interface Lexer.token
-  let toplevel_phrase = Parser.toplevel_phrase Lexer.token
-  let use_file = Parser.use_file Lexer.token
+  let implementation lexbuf =
+    From_current.copy_structure (Parser.implementation Lexer.token lexbuf)
+
+  let core_type lexbuf =
+    From_current.copy_core_type
+      (Parser.parse_core_type Lexer.token lexbuf)
+
+  let interface lexbuf =
+    From_current.copy_signature
+      (Parser.interface Lexer.token lexbuf)
+
+  let toplevel_phrase lexbuf =
```

Comment:
It took me a while to realize why you have these `From_current` transforms for the OCaml syntax but not the "JS" syntax below. I realized it's because the OCaml parser parses into Current, and the Reason one parses into 4.04. A small comment mentioning that would be nice.